enhancements/storage/csi-vsphere-operator.md

```diff
@@ -0,0 +1,172 @@
+* Enable creation of vSphere CSI storageclass, so as OCP users can start consuming vSphere CSI volumes without requiring further configuration.
+
+### Non-Goals
+
+* We don't intend to create a brand new driver with this KEP but merely want to deploy and configure upstream vSphere CSI driver - https://github.com/kubernetes-sigs/vsphere-csi-driver/
+
+
+## Proposal
+
+OCP ships with a vsphere-csi-driver-operator by default which is managed by [cluster-storage-operator](https://github.com/openshift/cluster-storage-operator/).
+vSphere CSI driver has few dependencies on installer though and they are:
+
+### Installer dependency
+
+* vSphere CSI driver requires HW version 15 on VMs that make up OCP cluster. Currently the rhcos OVA file Red Hat ships has default HW version configured to 13 and hence VM version should be updated.
+* Configuration of vSphere StorageClass requires knowledge of storage policy that was created in vCenter. Without this information - it is not possible to create a working storageClass for CSI driver.
+
+#### HW version of OCP VMs
+
+As mentioned above - vSphere CSI driver requires HW version 15 on all the VMs that make OCP cluster. Since, Openshift defaults to HW version 13 when a vSphere OCP cluster gets created - vSphere CSI driver isn't workable on the OCP cluster by default.
+
+To solve this following alternatives were considered:
+
+* We could provide manual instructions to the user about updating hardware version of their VMs. This won't be automatic but in some cases not even possible because some existing VMs can't be upgraded to HW version 15 in-place and hence this option is ruled out.
+* We could update installer to set HW version 15 - when appropriate hypervisor version(6.7u3 or higher) is detected.
+* For UPI install - we can additionally document required HW version while creating VMs.
+
+### Deployment strategy
+
+vSphere CSI driver operator will be deployed by [cluster-storage-operator](https://github.com/openshift/cluster-storage-operator/)
+
+#### Operator deployment via cluster-storage-operator
+
+The cluster-storage-operator will deploy all the resources necessary for creating the vSphere CSI driver operator.
+
+1. vSphere CSI driver operator will be deployed in namespace `openshift-cluster-csi-drivers`.
+2. A service account will be created for running the operator.
+3. The operator will get RBAC rules necessary for running the operator and its operand (i.e the actual vSphere CSI driver).
+4. A deployment will be created for starting the operator and will be managed by cluster-storage-operator.
+5. A instance of `ClusterCSIDriver` will be created to faciliate managment of driver operator.  `ClusterCSIDriver` is already defined in - https://github.com/openshift/api/blob/master/operator/v1/types_csi_cluster_driver.go but needs to be expanded to include vSphere CSI driver.
+6. cluster-storage-operator will create required cloud-credentials for talking with vCenter API.
```

Comment:
Unfortunately `cluster-storage-operator` can't create those creds, they should be created in `cluster-version-operator` regardless of the platform. `cloud-credentials-operator` might be running in manual mode and users should be able to create the creds manually.

Reply:
What about the StorageClass?

Reply:
Yes by CSO I meant, it is CSO that will carry that request and ensure that CVO creates them. How we do for other cloudproviders. My intention is exactly what you say but wording needs fixing I think.

Also for storageClass - SC will be created by the operator itself.